File: app/src/main/java/com/marknkamau/justjava/ui/drinkdetails/DrinkDetailsActivityPresenter.java
package com.marknkamau.justjava.ui.drinkdetails;

import android.content.SharedPreferences;

import com.marknkamau.justjava.models.CartItem;
import com.marknkamau.justjava.utils.Constants;
import com.marknkamau.justjava.utils.FirebaseAuthUtils;
import com.marknkamau.justjava.utils.RealmUtils;

class DrinkDetailsActivityPresenter {
    private DrinkDetailsActivityView activityView;
    private SharedPreferences sharedPreferences;

    DrinkDetailsActivityPresenter(DrinkDetailsActivityView activityView, SharedPreferences sharedPreferences) {
        this.activityView = activityView;
        this.sharedPreferences = sharedPreferences;
    }

    void addToCart(CartItem cartItem){
        new RealmUtils().saveNewItem(cartItem);
        activityView.displayMessage("Item added to cart");
        activityView.finishActivity();
    }

    void logUserOut(){
        FirebaseAuthUtils.INSTANCE.logOut();
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.remove(Constants.INSTANCE.getDEF_NAME());
        editor.remove(Constants.INSTANCE.getDEF_PHONE());
        editor.remove(Constants.INSTANCE.getDEF_ADDRESS());
        editor.apply();
    }
}

File: app/src/main/java/com/marknkamau/justjava/ui/drinkdetails/DrinkDetailsActivityPresenter.kt
package com.marknkamau.justjava.ui.drinkdetails

import android.content.SharedPreferences

import com.marknkamau.justjava.models.CartItem
import com.marknkamau.justjava.utils.Constants
import com.marknkamau.justjava.utils.FirebaseAuthUtils
import com.marknkamau.justjava.utils.RealmUtils

internal class DrinkDetailsActivityPresenter(private val activityView: DrinkDetailsActivityView, private val sharedPreferences: SharedPreferences) {

    fun addToCart(cartItem: CartItem) {
        RealmUtils().saveNewItem(cartItem)
        activityView.displayMessage("Item added to cart")
        activityView.finishActivity()
    }

    fun logUserOut() {
        FirebaseAuthUtils.logOut()
        val editor = sharedPreferences.edit()
        editor.remove(Constants.DEF_NAME)
        editor.remove(Constants.DEF_PHONE)
        editor.remove(Constants.DEF_ADDRESS)
        editor.apply()
    }
}
